AppInspector.CLI/Program.cs

```diff
@@ -25,103 +26,364 @@ static int Main(string[] args)
             WriteOnce.Verbosity = WriteOnce.ConsoleVerbosity.Medium;
             try
             {
-                var argsResult = Parser.Default.ParseArguments<AnalyzeCommandOptions,
-                    TagDiffCommandOptions,
-                    TagTestCommandOptions,
-                    ExportTagsCommandOptions,
-                    VerifyRulesCommandOptions,
-                    PackRulesCommandOptions>(args)
+                var argsResult = Parser.Default.ParseArguments<CLIAnalyzeCmdOptions,
+                    CLITagDiffCmdOptions,
+                    CLITagTestCmdOptions,
+                    CLIExportTagsCmdOptions,
+                    CLIVerifyRulesCmdOptions,
+                    CLIPackRulesCmdOptions>(args)
                   .MapResult(
-                    (AnalyzeCommandOptions opts) => RunAnalyzeCommand(opts),
-                    (TagDiffCommandOptions opts) => RunTagDiffCommand(opts),
-                    (TagTestCommandOptions opts) => RunTagTestCommand(opts),
-                    (ExportTagsCommandOptions opts) => RunExportTagsCommand(opts),
-                    (VerifyRulesCommandOptions opts) => RunVerifyRulesCommand(opts),
-                    (PackRulesCommandOptions opts) => RunPackRulesCommand(opts),
+                    (CLIAnalyzeCmdOptions cliOptions) => VerifyOutputArgsRun(cliOptions),
```

Comment:
Why do you have separate options for the CLI and the Library?  It looks like they overlap nearly entirely.

Reply:
Close but for the OutputFormat and OutputFilePath which are only applicable for CLI operations not for NuGet callers where file output is not supported in lieu of returning objects that they can serialize to file as needed.  In addition, the CLI version is not inherited from the options set for Commands module since that would require liquidhtml dependency where it is not used/needed so that the right property markers could be added required for the CommandLineParser used only in the CLI.  The solution was to transfer only the required arguments to the CommandOptions used in the Commands module allowing the CLI to verify the output arguments separately and manage file output.